File: chrome/browser/password_manager/android/test_support/java/src/org/chromium/chrome/browser/password_manager/PasswordManagerTestUtilsBridge.java
// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.password_manager;

import org.jni_zero.CalledByNativeForTesting;
import org.jni_zero.NativeMethods;

/**
 * This bridge contains static methods related to password manager test setup. It's intended only to
 * be used in tests.
 */
public class PasswordManagerTestUtilsBridge {
    /** Disables server predictions to speed up tests */
    public static void disableServerPredictions() {
        PasswordManagerTestUtilsBridgeJni.get().disableServerPredictions();
    }

    @CalledByNativeForTesting
    public static void setUpGmsCoreFakeBackends() {
        PasswordManagerTestHelper.setUpGmsCoreFakeBackends();
    }

    @NativeMethods
    interface Natives {
        void disableServerPredictions();
    }
}

File: chrome/browser/password_manager/android/password_manager_test_utils_bridge.cc
// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <jni.h>

#include "base/android/jni_android.h"
#include "components/password_manager/core/browser/password_form_manager.h"

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "chrome/browser/password_manager/android/test_support_jni_headers/PasswordManagerTestUtilsBridge_jni.h"

void JNI_PasswordManagerTestUtilsBridge_DisableServerPredictions(JNIEnv* env) {
  password_manager::PasswordFormManager::
      DisableFillingServerPredictionsForTesting();
}

void SetUpGmsCoreFakeBackends() {
  Java_PasswordManagerTestUtilsBridge_setUpGmsCoreFakeBackends(
      jni_zero::AttachCurrentThread());
}
